addons/stock/wizard/stock_picking_return.py

```diff
@@ -34,17 +34,17 @@ def default_get(self, fields):
         return res
 
     picking_id = fields.Many2one('stock.picking')
-    product_return_moves = fields.One2many('stock.return.picking.line', 'wizard_id', 'Moves')
-    move_dest_exists = fields.Boolean('Chained Move Exists', readonly=True)
-    original_location_id = fields.Many2one('stock.location')
-    parent_location_id = fields.Many2one('stock.location')
+    product_return_moves = fields.One2many('stock.return.picking.line', 'wizard_id', 'Moves', compute='_compute_moves_locations', readonly=False, store=True)
+    move_dest_exists = fields.Boolean('Chained Move Exists', readonly=True, compute='_compute_moves_locations', store=True)
+    original_location_id = fields.Many2one('stock.location', compute='_compute_moves_locations', store=True)
+    parent_location_id = fields.Many2one('stock.location', compute='_compute_moves_locations', store=True)
     company_id = fields.Many2one(related='picking_id.company_id')
     location_id = fields.Many2one(
-        'stock.location', 'Return Location',
+        'stock.location', 'Return Location', compute='_compute_moves_locations', readonly=False, store=True,
         domain="['|', ('id', '=', original_location_id), '|', '&', ('return_location', '=', True), ('company_id', '=', False), '&', ('return_location', '=', True), ('company_id', '=', company_id)]")
 
-    @api.onchange('picking_id')
-    def _onchange_picking_id(self):
+    @api.depends('picking_id')
+    def _compute_moves_locations(self):
         move_dest_exists = False
```

Comment:
probably needs a `for wizard in self:` + the rest of the method should be updated to match

onchanges = always 1 record, compute = possibly multiple records